starter-code/routes/index.js

```diff
@@ -0,0 +1,106 @@
+/*jshint esversion: 6 */
+
+const express      = require('express');
+const router       = express.Router();
+const User         = require("../models/user");
+const bcrypt       = require("bcrypt");
+const saltRounds   = 10;
+const session      = require("express-session");
+const MongoStore   = require("connect-mongo")(session);
+
+/* GET home page. */
+
+router.get('/', (req, res, next) => {
+  if (req.session.currentUser) {
+    res.redirect('welcome');
+  } else {
+    res.render('sign-up/index');
+  }
+});
+
+router.get('/welcome', (req, res, next) => {
+  if (!req.session.currentUser) {
+    res.redirect('/');
+  } else {
+  res.render('welcome');
+  }
+});
+
+router.post('/', (req, res, next) => {
+
+    const newUser = User({
```

Comment:
you should also authorize this route like you did with the `.get('/')` - only anonymous users should be able to POST the form